catt/controllers.py

```diff
@@ -22,29 +23,60 @@ def get_stream_info(video_url):
     return stream_info
 
 
+class ChromecastDeviceError(ClickException):
+    pass
+
+
 class Cache:
-    def __init__(self, cache_dir=os.path.join(tempfile.gettempdir(), "catt_cache/")):
+    def __init__(self, duration=3 * 24 * 3600,
+                 cache_dir=os.path.join(tempfile.gettempdir(), "catt_cache")):
         self.cache_dir = cache_dir
         try:
             os.mkdir(cache_dir)
         except:
             pass
 
-    def _get_cache_filename(self, key):
-        return os.path.join(self.cache_dir, key)
+        cache_filename = os.path.join(cache_dir, "chromecast_hosts")
+        self.cache_filename = cache_filename
 
-    def get(self, key, duration):
-        cache_filename = self._get_cache_filename(key)
         if os.path.exists(cache_filename):
-            if os.path.getctime(cache_filename) + duration > time.time():
-                return open(cache_filename).read()
-            else:
-                os.remove(cache_filename)
+            if os.path.getctime(cache_filename) + duration < time.time():
+                self._initialize_cache()
         else:
+            self._initialize_cache()
+
+    def _initialize_cache(self):
+        data = {}
+        devices = pychromecast.get_chromecasts()
+        for device in devices:
+            data[device.name] = device.host
+        self._write_cache(data)
+
+    def _read_cache(self):
```

Comment:
We can probably store the loaded/written data in this instance in the future, but for now we're only calling `.get` once, so it's fine.